app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -551,10 +551,24 @@ def get_mac_address_of_nic_on_requested_vlan(args)
     end
 
     def find_mac_address_on_network(nics, vnic_profile_id)
-      nic = nics.detect do |n|
-        n.vnic_profile.id == vnic_profile_id
+      if vnic_profile_id == '<Empty>'
```

Comment:
please extract '<Empty>' to constant - it is used widely and we should define it in one place, so let's start with defining a constant for it in this PR, and in a follow PR replace other places.